### PR TITLE
Hops

### PR DESCRIPTION
**Pull request by jfy133**
This completes the entire useful HOPS pipeline (MALT and MaltExtract).

It also includes a few extra under-the-hood changes (improving github actions to stop nf-tower issues, version bump for SexDet.).

Note: Travis WILL partially fail because I use the nextflow 19.10.0 `path` output qualifer which is not avaliable in 19.04.0 (which is still the minimum version in travis). @apeltzer should we bump this minium version now?

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If necessary, also make a PR on the [nf-core/eager branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/eager)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: .github/workflows/nf-core_eager.yml [hunk 1]
```diff
@@ -88,18 +88,34 @@ jobs:
...
+          mkdir -p databases/malt
+          readlink -f databases/malt/
+          for i in index0.idx ref.db ref.idx ref.inf table0.db table0.idx taxonomy.idx taxonomy.map taxonomy.tre; do wget https://github.com/nf-core/test-datasets/raw/eager/databases/malt/"$i" -P databases/malt/; done
+      - name: METAGENOMIC Run the basic pipeline but with unmapped reads going into MALT
+        run: |
+          nextflow run ${GITHUB_WORKSPACE} "$TOWER" -name "$RUN_NAME-malt" -profile test,docker --pairedEnd --run_bam_filtering --bam_discard_unmapped --bam_unmapped_type 'fastq' --run_metagenomic_screening --database "/home/runner/work/eager/eager/databases/malt/"
+      - name: MALTEXTRACT Download resource files
+        run: |
+            mkdir -p databases/maltextract
+            for i in ncbi.tre ncbi.map; do wget https://github.com/rhuebler/HOPS/raw/external/Resources/"$i" -P databases/maltextract/; done
```
Comment (maxibor) [inline .github/workflows/nf-core_eager.yml]: Could you change this to use the tagged version of the latest HOPS release ?
`https://github.com/rhuebler/HOPS/raw/0.33/Resources/`

Comment (jfy133) [inline .github/workflows/nf-core_eager.yml]:
```suggestion
            for i in ncbi.tre ncbi.map; do wget https://github.com/rhuebler/HOPS/raw/0.33/Resources/"$i" -P databases/maltextract/; done
```

**Comment by apeltzer** [comment]
```
Warning, pipeline completed, but with errored process(es) 

Number of ignored errored process(es) : 0 

Number of successfully ran process(es) : 23 
```

This keeps annoying me. Can we also fix this in this PR? 
It's just changing this here (and the following lines):

https://github.com/nf-core/eager/blob/b8d3dec3f044da5ec016493108910945be2b8702/main.nf#L1431

to 

```
if (workflow.stats.ignoredCount > 0 && workflow.success) {
      log.info "${c_purple}Warning, pipeline completed, but with errored process(es) ${c_reset}"
      log.info "${c_red}Number of ignored errored process(es) : ${workflow.stats.ignoredCount} ${c_reset}"
      log.info "${c_green}Number of successfully ran process(es) : ${workflow.stats.succeedCount} ${c_reset}"
    }

    if (workflow.success) {
        log.info "${c_purple}[{{ cookiecutter.name }}]${c_green} Pipeline completed successfully${c_reset}"
    } else {
        checkHostname()
        log.info "${c_purple}[{{ cookiecutter.name }}]${c_red} Pipeline completed with errors${c_reset}"
    }
```

**Comment by jfy133** [comment]
> if (workflow.stats.ignoredCount > 0 && workflow.success) {
>       log.info "${c_purple}Warning, pipeline completed, but with errored process(es) ${c_reset}"
>       log.info "${c_red}Number of ignored errored process(es) : ${workflow.stats.ignoredCount} ${c_reset}"
>       log.info "${c_green}Number of successfully ran process(es) : ${workflow.stats.succeedCount} ${c_reset}"
>     }
> 
>     if (workflow.success) {
>         log.info "${c_purple}[{{ cookiecutter.name }}]${c_green} Pipeline completed successfully${c_reset}"
>     } else {
>         checkHostname()
>         log.info "${c_purple}[{{ cookiecutter.name }}]${c_red} Pipeline completed with errors${c_reset}"
>     }

God yes, that's being annoying me too. Incoming!

File: main.nf [hunk 2]
```diff
@@ -2071,17 +2293,17 @@ workflow.onComplete {
...
-      log.info "${c_red}Number of ignored errored process(es) : ${workflow.stats.ignoredCountFmt} ${c_reset}"
-      log.info "${c_green}Number of successfully ran process(es) : ${workflow.stats.succeedCountFmt} ${c_reset}"
+      log.info "${c_red}Number of ignored errored process(es) : ${workflow.stats.ignoredCount} ${c_reset}"
+      log.info "${c_green}Number of successfully ran process(es) : ${workflow.stats.succeedCount} ${c_reset}"
     }
 
-    if(workflow.success){
-        log.info "${c_purple}[nf-core/eager]${c_green} Pipeline completed successfully${c_reset}"
+    if (workflow.success) {
+        log.info "${c_purple}[{{ cookiecutter.name }}]${c_green} Pipeline completed successfully${c_reset}"
```
Comment (apeltzer) [inline main.nf]:
```suggestion
        log.info "${c_purple}nf-core/eager${c_green} Pipeline completed successfully${c_reset}"
```

File: main.nf [hunk 3]
```diff
@@ -2071,17 +2293,17 @@ workflow.onComplete {
...
     } else {
         checkHostname()
-        log.info "${c_purple}[nf-core/eager]${c_red} Pipeline completed with errors${c_reset}"
+        log.info "${c_purple}[{{ cookiecutter.name }}]${c_red} Pipeline completed with errors${c_reset}"
```
Comment (apeltzer) [inline main.nf]:
```suggestion
        log.info "${c_purple}nf-core/eager${c_red} Pipeline completed with errors${c_reset}"
```

File: main.nf [hunk 4]
```diff
@@ -312,6 +339,36 @@ where_are_my_files = file("$baseDir/assets/where_are_my_files.txt")
...
+// maltextract - only including number 
+// parameters if default documented or duplicate of MALT
+params.run_maltextract = false
+params.maltextract_taxon_list = ''
+params.maltextract_ncbifiles = ''
+params.maltextract_filter = "def_anc"
+params.maltextract_toppercent = 0.01
+params.maltextract_destackingoff = false
+params.maltextract_downsamplingoff = false
+params.maltextract_duplicateremovaloff = false
```
Comment (apeltzer) [inline main.nf]: All of these params should be in `nextflow.config` - I'll do that with the TSV branch unless you want to...

Comment (jfy133) [reply main.nf]: As in all default paramaters? I can clean that up this evening.

**Pull request merged**